report/main.qmd

```diff
@@ -1 +1,166 @@
+#missing data, overall
+plot_missing_values_proportions(df=sale_data) +
+  ggtitle("", subtitle = "Figure 1: The proportion of missing values for all columns in \nthe dataset.")
+```
+
+```{r missing-data-table}
+# data missingness  by Site
+count_missing<- sale_data %>%
+    group_by(Site) %>%
+    summarise(across(.cols = where(~ any(is.na(.x))), ~sum(is.na(.x))))
+
+# Table
+kable(count_missing, caption = "Number of missing observations for columns with missing data present by site")
+```
+
+To explore the change of daily sales of zero-calorie beverages and bottled sugared beverages over time, time series plot for `ZeroCal` and `Sugary` can be made for each of the three sites. It is important to note that the increase in the raw counts of say zero-calorie beverages being sold does not necessarily mean the decrease in the sales of sugared beverages. It could be the case that more customs came to the shops that day so that sales of all things tended to be higher. Therefore, in addition to visualize the change of daily sales using the raw counts, it is even more important to explore the change of daily sales in percentage terms (i.e. `ZeroCal`/(`ZeroCal`+`Sugary`)\*100\% and `Sugary`/(`ZeroCal`+`Sugary`)\*100\%)
+(see Figure 2 as an example). Change in the zero-calorie beverages percentage can better represent the change in zero-calorie beverages consumption relative to the sugared beverages consumption, and vice versa. 
+
+```{r}
+# delete rows with missing data in both ZeroCal and Sugary
+sale_data_c <- sale_data %>%
+  filter(!(is.na(ZeroCal) | is.na(Sugary)))
+```
+
+To compare the consumption of a selected type of beverages (zero-calorie beverages or sugary beverages) under different `intervention` categories, boxplots with sales in percentage terms as y-axis and with `intervention` as x-axis can be made (see Figure 3 as an example). Similarly, boxplots can also be used to compare the consumption of a selected type of beverages among days of week to see if there are any patterns.
+\
+```{r time-series-plot, fig.height=3.8, fig.width=7}
+sale_data_c$Sugary_percent <-
+  sale_data_c$Sugary/(sale_data_c$Sugary+sale_data_c$ZeroCal)*100
+sale_data_c$ZeroCal_percent <-
+  sale_data_c$ZeroCal/(sale_data_c$Sugary+sale_data_c$ZeroCal)*100
+
+# Plot Daily sales of ZeroCal_percent
+ggplot(data=sale_data_c, 
+       aes(x=Count, y=ZeroCal_percent, group=Site)) +
+  geom_line(aes(color=Intervention)) + 
+  geom_point(aes(color=Intervention), size=0.5) + 
+  facet_wrap( ~ Site, ncol = 1, scales = "free_y") + 
+  xlab("Day") + 
+  ylab("Daily percentage of \nzero-calorie beverages sales (%)") +
+  ggtitle("",subtitle = "Figure 2: Daily percentage of zero-calorie beverages sales over time for \neach of the three sites")
+```
+
+
+```{r boxplot}
+#Create a new Intervention column to make boxplots
+sale_data_c$Intervention_new <- sale_data_c$Intervention
+# replace the "preint" in the 2nd half of the study with "preint2"
```

Comment:
Instead of faceting off site, it might make comparisons easier if you have one plot, with side-by-side box-plots for each site. This way, you can directly compare each intervention across sites